pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -11,12 +11,42 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import os
 from abc import ABC, abstractmethod
+from typing import Dict
+
+from pytorch_lightning.utilities import rank_zero_warn
 
 
 class ClusterEnvironment(ABC):
     """ Specification of a cluster environment. """
 
+    DEFAULT_ENVIRON_SETTINGS = {
+        # these are default NCCL settings for communication speedup
+        "NCCL_NSOCKS_PERTHREA": "4",
+        "NCCL_SOCKET_NTHREADS": "2",
+    }
+
+    def __init__(self, environ_settings: Dict[str, str] = {}):
```

Comment:
@awaelchli, do you suggest that ClusterEnvironment might not be the best place to override these parameters?
i.e. not exposing `environ_settings`, we just apply the good default value, if application users would like to override, they could update explicitly before launching Lightning trainer.

Reply:
fine with not exposing this arg, the main point of this PR is to set some good default so that most users do not need to worry about this but also benefit the speedup, "power" users could still override these.

Reply:
open to discussion whether we would like to override these inside Trainer or not.

Reply:
>  do you suggest that ClusterEnvironment might not be the best place to override these parameters?

I think it's not a bad place :) Another option would be in the GPU accelerator since NCCL is used only when we communicate with GPUs right? Either way I think having good defaults is great to have there! I vote for not exposing the environ_settings though.

Reply:
I agree with @awaelchli - the cluster environment is more of a read-only dataclass. it also means that extensions of the cluster environment need to call `super().__init_(environ_settings)` which feels out of place with this.

I prefer keeping these nccl settings in the training plugins that use them instead, and not worrying about general environment variable updates since the application could control this before invoking the trainer function

Reply:
so the consensus is that application will control environment variable update outside trainer, and inside trainer, "relevant" training plugins will set good default. 

will update correspondingly.